traffic_ops/app/lib/API/Configs/ApacheTrafficServer.pm

```diff
@@ -0,0 +1,1959 @@
+			{
+				$regex_time{ $row->asset_url } = $purge_end;
+			}
+		}
+	}
+
+	foreach my $re ( sort keys %regex_time ) {
+		$text .= $re . " " . $regex_time{$re} . "\n";
+	}
+
+	return $text;
+}
+
+sub set_dscp_dot_config {
+	my $self     = shift;
+	my $cdn_obj  = shift;
+	my $filename = shift;
+	my $scope    = shift;
+
+	my $text = $self->header_comment( $cdn_obj->name );
+	my $dscp_decimal;
+	if ( $filename =~ /^set_dscp_(\d+)\.config$/ ) {
+		$dscp_decimal = $1;
+	}
+	else {
+		$text = "An error occured generating the DSCP header rewrite file.";
+	}
+	$text .= "cond %{REMAP_PSEUDO_HOOK}\n" . "set-conn-dscp " . $dscp_decimal . " [L]\n";
+
+	return $text;
+}
+
+sub ssl_multicert_dot_config {
+	my $self     = shift;
+	my $cdn_obj  = shift;
+	my $filename = shift;
+	my $scope    = shift;
+
+	my $text = $self->header_comment( $cdn_obj->name );
+
+	## We should break this search out into a separate sub later
+	my $protocol_search = '> 0';
+	my @ds_list = $self->db->resultset('Deliveryservice')->search( { -and => [ cdn_id => $cdn_obj->id, 'me.protocol' => \$protocol_search ] } )->all();
+	foreach my $ds (@ds_list) {
+		my $ds_id        = $ds->id;
+		my $xml_id       = $ds->xml_id;
+		my $rs_ds        = $self->db->resultset('Deliveryservice')->search( { 'me.id' => $ds_id } );
+		my $data         = $rs_ds->single;
+		my $domain_name  = UI::DeliveryService::get_cdn_domain( $self, $ds_id );
+		my $ds_regexes   = UI::DeliveryService::get_regexp_set( $self, $ds_id );
+		my @example_urls = UI::DeliveryService::get_example_urls( $self, $ds_id, $ds_regexes, $data, $domain_name, $data->protocol );
+
+		#first one is the one we want
+		my $hostname = $example_urls[0];
+		$hostname =~ /(https?:\/\/)(.*)/;
+		my $new_host = $2;
+		my $key_name = "$new_host.key";
+		$new_host =~ tr/./_/;
+		my $cer_name = $new_host . "_cert.cer";
+
+		$text .= "ssl_cert_name=$cer_name\t ssl_key_name=$key_name\n";
+	}
+
+	return $text;
+}
+
+sub url_sig_dot_config {
+	my $self        = shift;
+	my $profile_obj = shift;
+	my $filename    = shift;
+	my $scope       = shift;
+
+	my $sep = defined( $separator->{$filename} ) ? $separator->{$filename} : " = ";
+	my $data = $self->profile_param_data( $profile_obj->id, $filename );
+	my $text = $self->header_comment( $profile_obj->name );
+
+	my $response_container = $self->riak_get( URL_SIG_KEYS_BUCKET, $filename );
+	my $response = $response_container->{response};
+	if ( $response->is_success() ) {
+		my $response_json = decode_json( $response->content );
+		my $keys          = $response_json;
+		foreach my $parameter ( sort keys %{$data} ) {
+			if ( !defined($keys) || $parameter !~ /^key\d+/ ) {    # only use key parameters as a fallback (temp, remove me later)
+				$text .= $parameter . $sep . $data->{$parameter} . "\n";
+			}
+		}
+
+		foreach my $parameter ( sort keys %{$keys} ) {
+			$text .= $parameter . $sep . $keys->{$parameter} . "\n";
+		}
+
+		return $text;
+	}
+	else {
+		return;
```

Comment:
do you want to return nothing? or do you want to return $text which at this point only contains the header_comment? if nothing, maybe kill the "else" and just add return undef to be more explicit. or just kill the else entirely.